src/packet/route/route.rs

```diff
@@ -283,6 +289,58 @@ impl<'a> ToPayload for RtAttrPacket<'a> {
     }
 }
 
+impl<'a> ToPayload for &'a [&'a Nested<'a>] {
+    fn payload_add(&self, payload: &mut [u8]) {
+        self.iter().fold(0, |pos, pkg| {
+            pkg.payload_add(&mut payload[pos..]);
+            pos + pkg.payload_size()
+        });
+    }
+
+    fn payload_size(&self) -> usize {
+        self.iter().map(|p| p.payload_size()).sum()
+    }
+}
+
+impl<'a> ToPayload for &'a [u8] {
+    fn payload_add(&self, payload: &mut [u8]) {
+        payload[..self.len()].copy_from_slice(self)
+    }
+
+    fn payload_size(&self) -> usize {
+        self.len() + 1
```

Comment:
Why plus one? I would expect that the length is encoded as is, don't know why it should have a NullByte at the end.